src/Component.php

```diff
@@ -193,13 +193,16 @@ protected function getVariantDefaultVariables(string $modifier, array $defined_v
    *   2. the component basename without variants
    *   3. a list of variants, if any.
    */
-  protected function extractParts(string $pathname): array {
+  protected function extractParts(string $compound_name): array {
     $library = Drupal::service('twig.loader.componentlibrary');
-    $variants = explode('--', basename(basename($pathname, '.twig'), '.html'));
+    $variants = explode('--', basename(basename($compound_name, '.twig'), '.html'));
     $component = array_shift($variants);
     // If no file-based variant template file exist use the base component otherwise.
-    if (!$library->exists($pathname)) {
-      $pathname = preg_replace('@--[^.]+@', '', $pathname);
+    if (!$library->exists($compound_name)) {
```

Comment:
For readability, I decided to go with a full if/else statement over a ternary. Also I wasn't very happy with declaring the `$pathname` variable before and have just a if condition instead.